types/mongodb/index.d.ts

```diff
@@ -38,59 +38,109 @@ export class MongoError extends Error {
+    
+    // If the database authentication is dependent on another databaseName.
     authSource?: string;
-    //  the write concern for the operation where < 1 is no acknowlegement of write and w >= 1, w = ‘majority’ or tag acknowledges the write.
+    // Default: null;https://docs.mongodb.com/manual/reference/write-concern/#write-concern
     w?: number | string;
-    // set the timeout for waiting for write concern to finish (combines with w option).
+    // The write concern timeout to finish (combining with w option).
     wtimeout?: number;
+    // Specify a journal write concern.
     j?: boolean;
-    // use c++ bson parser. default:false.
-    native_parser?: boolean;
-    // force server to create _id fields instead of client. default:false.
+    // Force server to create _id fields instead of client.
     forceServerObjectId?: boolean;
+    // Use c++ bson parser. .
+    native_parser?: boolean;
+    // Serialize functions on any object.
     serializeFunctions?: boolean;
+    // Specify if the BSON serializer should ignore undefined fields.
     ignoreUndefined?: boolean;
-    // peform operations using raw bson buffers. default:false.
+    // Return document results as raw BSON buffers.
     raw?: boolean;
-    // when deserializing a Long will fit it into a Number if it’s smaller than 53 bits. default:true.
+    // Default: true; Promotes Long values to number if they fit inside the 53 bits resolution.
     promoteLongs?: boolean;
-    bufferMaxEntries?: number;
+    // Default: -1 (unlimited); Amount of operations the driver buffers up untill discard any new ones
+    promoteBuffers?: number;
     // the prefered read preference. use 'ReadPreference' class.
     readPreference?: ReadPreference | string;
-    // custom primary key factory to generate _id values (see Custom primary keys).
+    // Default: true; Promotes BSON values to native types where possible, set to false to only receive wrapper types.
+    promoteValues?: Object;
+    // Custom primary key factory to generate _id values (see Custom primary keys).
     pkFactory?: Object;
+    // ES6 compatible promise constructor
     promiseLibrary?: Object;
+    // https://docs.mongodb.com/manual/reference/read-concern/#read-concern
     readConcern?: { level?: Object };
 }
 
-// See http://mongodb.github.io/node-mongodb-native/2.1/api/ReadPreference.html
-export class ReadPreference {
-    constructor(mode: string, tags: Object);
-    mode: string;
-    tags: any;
-    static PRIMARY: string;
-    static PRIMARY_PREFERRED: string;
-    static SECONDARY: string;
-    static SECONDARY_PREFERRED: string;
-    static NEAREST: string;
-    isValid(mode: string): boolean;
-    static isValid(mode: string): boolean;
-}
-
-//http://mongodb.github.io/node-mongodb-native/2.1/api/Server.html
-export interface SocketOptions {
+// http://mongodb.github.io/node-mongodb-native/2.2/api/Server.html
+export interface SocketOptions extends SSLOptions, HighAvailabilityOptions{
```

Comment:
`SocketOption` should not extend `SSLOptions`, `HighAvailabilityOptions` (at least according to the documentation here http://mongodb.github.io/node-mongodb-native/2.2/api/Server.html)